crates/frontend/src/components/default_config_form.rs

```diff
@@ -197,52 +197,117 @@ where
             <div class="divider"></div>
 
             {move || {
-                config_type_rs
-                    .with(|c_type| {
-                        let input_format = match c_type.as_str() {
-                            "Number" | "Decimal" => {
-                                view! {
-                                    <input
-                                        type="number"
-                                        placeholder="Value"
-                                        class="input input-bordered w-full max-w-md"
-                                        value=config_value.get()
-                                        on:change=move |ev| {
-                                            logging::log!("{:?}", event_target_value(& ev));
-                                            set_config_value.set(event_target_value(&ev));
-                                        }
-                                    />
-                                }
-                                    .into_view()
-                            }
-                            _ => {
-                                view! {
-                                    <textarea
-                                        type="text"
-                                        placeholder="Value"
-                                        class="input input-bordered w-full max-w-md"
-                                        on:change=move |ev| {
-                                            logging::log!("{:?}", event_target_value(& ev));
-                                            set_config_value.set(event_target_value(&ev));
-                                        }
-                                    >
-
-                                        {config_value.get()}
-                                    </textarea>
+                let schema: Map<String, Value> = serde_json::from_value(config_schema_rs.get())
```

Comment:
Should deciding input field based on schema type be a component of it's own? It is used across 3 files and should behave the same for all of them

Reply:
Sure.